When using buildpack without docker running gets a meaningless exception

### Description

When using buildpack without docker running gets a meaningless exception:

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.1.Final:build (default) on project hello: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.buildpack.deployment.BuildpackProcessor#buildFromJar threw an exception: java.lang.RuntimeException: java.io.IOException: com.sun.jna.LastErrorException: [2] No such file or directory
[ERROR] 	at com.github.dockerjava.httpclient5.ApacheDockerHttpClientImpl.execute(ApacheDockerHttpClientImpl.java:187)
[ERROR] 	at com.github.dockerjava.httpclient5.ApacheDockerHttpClient.execute(ApacheDockerHttpClient.java:9)
[ERROR] 	at com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:228)
[ERROR] 	at com.github.dockerjava.core.DefaultInvocationBuilder.get(DefaultInvocationBuilder.java:202)
[ERROR] 	at com.github.dockerjava.core.DefaultInvocationBuilder.get(DefaultInvocationBuilder.java:74)
[ERROR] 	at com.github.dockerjava.core.exec.ListImagesCmdExec.execute(ListImagesCmdExec.java:41)
[ERROR] 	at com.github.dockerjava.core.exec.ListImagesCmdExec.execute(ListImagesCmdExec.java:16)
[ERROR] 	at com.github.dockerjava.core.exec.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:21)
[ERROR] 	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
[ERROR] 	at dev.snowdrop.buildpack.docker.ImageUtils.pullImages(ImageUtils.java:39)
[ERROR] 	at dev.snowdrop.buildpack.Buildpack.prep(Buildpack.java:218)
[ERROR] 	at dev.snowdrop.buildpack.Buildpack.build(Buildpack.java:102)
[ERROR] 	at dev.snowdrop.buildpack.Buildpack.<init>(Buildpack.java:96)
[ERROR] 	at dev.snowdrop.buildpack.EditableBuildpack.<init>(EditableBuildpack.java:16)
[ERROR] 	at dev.snowdrop.buildpack.BuildpackBuilder.build(BuildpackBuilder.java:72)
[ERROR] 	at io.quarkus.container.image.buildpack.deployment.BuildpackProcessor.runBuildpackBuild(BuildpackProcessor.java:188)
[ERROR] 	at io.quarkus.container.image.buildpack.deployment.BuildpackProcessor.buildFromJar(BuildpackProcessor.java:82)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:832)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.io.IOException: com.sun.jna.LastErrorException: [2] No such file or directory
[ERROR] 	at com.github.dockerjava.transport.DomainSocket.<init>(DomainSocket.java:63)
[ERROR] 	at com.github.dockerjava.transport.BsdDomainSocket.<init>(BsdDomainSocket.java:43)
```

You can get what's happening but for a newbie maybe a better message could be provided like No Docker Host.

### Implementation ideas

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -213,7 +213,7 @@
         <strimzi-oauth.version>0.15.0</strimzi-oauth.version>
         <strimzi-oauth.nimbus.version>9.41.1</strimzi-oauth.nimbus.version>
         <jose4j.version>0.9.6</jose4j.version>
-        <java-buildpack-client.version>0.0.6</java-buildpack-client.version>
+        <java-buildpack-client.version>0.0.12</java-buildpack-client.version>
         <org-crac.version>0.1.3</org-crac.version>
         <sshd-common.version>2.12.1</sshd-common.version>
         <mime4j.version>0.8.11</mime4j.version>

--- a/extensions/container-image/container-image-buildpack/deployment/src/main/java/io/quarkus/container/image/buildpack/deployment/BuildpackConfig.java
+++ b/extensions/container-image/container-image-buildpack/deployment/src/main/java/io/quarkus/container/image/buildpack/deployment/BuildpackConfig.java
@@ -14,14 +14,20 @@ public class BuildpackConfig {
     /**
      * The buildpacks builder image to use when building the project in jvm mode.
      */
-    @ConfigItem
-    public Optional<String> jvmBuilderImage;
+    @ConfigItem(defaultValue = "paketocommunity/builder-ubi-base:latest")
+    public String jvmBuilderImage;
 
     /**
-     * The buildpacks builder image to use when building the project in jvm mode.
+     * The buildpacks builder image to use when building the project in native mode.
      */
     @ConfigItem
     public Optional<String> nativeBuilderImage;
+
+    /**
+     * Should the builder image be 'trusted' (use creator lifecycle)
+     */
+    @ConfigItem
+    public Optional<Boolean> trustBuilderImage;
 
     /**
      * Environment key/values to pass to buildpacks.
@@ -39,16 +45,29 @@ public class BuildpackConfig {
     public Optional<String> runImage;
 
     /**
-     * Max pull timeout for builder/run images, in seconds
+     * Initial pull timeout for builder/run images, in seconds
      */
     @ConfigItem(defaultValue = "300")
     public Integer pullTimeoutSeconds;
 
     /**
+     * Increase pull timeout for builder/run images after failure, in seconds
+     */
+    @ConfigItem(defaultValue = "15")
+    public Integer pullTimeoutIncreaseSeconds;
+
+    /**
+     * How many times to retry an image pull after a failure
+     */
+    @ConfigItem(defaultValue = "3")
+    public Integer pullRetryCount;
+
+    /**
      * DOCKER_HOST value to use.
      *
-     * If not set, the env var DOCKER_HOST is used, if that is not set
-     * the value `unix:///var/run/docker.sock' (or 'npipe:///./pipe/docker_engine' for windows) is used.
+     * If not set, the env var DOCKER_HOST is used, if that is not set the platform will look for
+     * 'npipe:///./pipe/docker_engine' for windows, and `unix:///var/run/docker.sock' then
+     * `unix:///var/run/podman.sock` then `unix:///var/run/user/<uid>/podman/podman.sock` for linux
      */
     @ConfigItem
     public Optional<String> dockerHost;

--- a/extensions/container-image/container-image-buildpack/deployment/src/main/java/io/quarkus/container/image/buildpack/deployment/BuildpackProcessor.java
+++ b/extensions/container-image/container-image-buildpack/deployment/src/main/java/io/quarkus/container/image/buildpack/deployment/BuildpackProcessor.java
@@ -18,8 +18,9 @@ import com.github.dockerjava.api.async.ResultCallback;
 import com.github.dockerjava.api.model.AuthConfig;
 import com.github.dockerjava.api.model.PushResponseItem;
 
-import dev.snowdrop.buildpack.Buildpack;
-import dev.snowdrop.buildpack.BuildpackBuilder;
+import dev.snowdrop.buildpack.BuildConfig;
+import dev.snowdrop.buildpack.BuildConfigBuilder;
+import dev.snowdrop.buildpack.config.ImageReference;
 import dev.snowdrop.buildpack.docker.DockerClientUtils;
 import io.quarkus.container.image.deployment.ContainerImageConfig;
 import io.quarkus.container.image.deployment.util.NativeBinaryUtil;
@@ -181,33 +182,49 @@ public class BuildpackProcessor {
 
         if (buildContainerImage) {
             log.info("Initiating Buildpack build");
-            Buildpack buildpack = Buildpack.builder()
-                    .addNewFileContent(dirs.get(ProjectDirs.ROOT).toFile())
-                    .withFinalImage(targetImageName)
+            int exitCode = BuildConfig.builder()
+                    .addNewFileContentApplication(dirs.get(ProjectDirs.ROOT).toFile())
+                    .withOutputImage(new ImageReference(targetImageName))
+                    .withNewPlatformConfig()
                     .withEnvironment(envMap)
-                    .withLogLevel(buildpackConfig.logLevel)
-                    .withPullTimeoutSeconds(buildpackConfig.pullTimeoutSeconds)
+                    .endPlatformConfig()
+                    .withNewLogConfig()
                     .withLogger(new BuildpackLogger())
-                    .accept(BuildpackBuilder.class, b -> {
-
+                    .withLogLevel(buildpackConfig.logLevel)
+                    .endLogConfig()
+                    .withNewDockerConfig()
+                    .withPullRetryIncreaseSeconds(buildpackConfig.pullTimeoutIncreaseSeconds)
+                    .withPullTimeoutSeconds(buildpackConfig.pullTimeoutSeconds)
+                    .withPullRetryCount(buildpackConfig.pullRetryCount)
+                    .endDockerConfig()
+                    .accept(BuildConfigBuilder.class, b -> {
                         if (isNativeBuild) {
-                            buildpackConfig.nativeBuilderImage.ifPresent(i -> b.withBuilderImage(i));
+                            buildpackConfig.nativeBuilderImage.ifPresent(i -> b.withBuilderImage(new ImageReference(i)));
                         } else {
-                            buildpackConfig.jvmBuilderImage.ifPresent(i -> b.withBuilderImage(i));
+                            b.withBuilderImage(new ImageReference(buildpackConfig.jvmBuilderImage));
                         }
 
                         if (buildpackConfig.runImage.isPresent()) {
                             log.info("Using Run image of " + buildpackConfig.runImage.get());
-                            b.withRunImage(buildpackConfig.runImage.get());
+                            b.withRunImage(new ImageReference(buildpackConfig.runImage.get()));
                         }
+
                         if (buildpackConfig.dockerHost.isPresent()) {
                             log.info("Using DockerHost of " + buildpackConfig.dockerHost.get());
-                            b.withDockerHost(buildpackConfig.dockerHost.get());
+                            b.editDockerConfig().withDockerHost(buildpackConfig.dockerHost.get())
+                                    .endDockerConfig();
                         }
 
-                    }).build();
+                        if (buildpackConfig.trustBuilderImage.isPresent()) {
+                            log.info("Setting trusted image to " + buildpackConfig.trustBuilderImage.get());
+                            b.editPlatformConfig().withTrustBuilder(buildpackConfig.trustBuilderImage.get())
+                                    .endPlatformConfig();
+                        }
+                    })
+                    .build()
+                    .getExitCode();
 
-            if (buildpack.getExitCode() != 0) {
+            if (exitCode != 0) {
                 throw new IllegalStateException("Buildpack build failed");
             }
 

